api/uaa/refresh_token.go

```diff
@@ -36,6 +51,7 @@ func (client *Client) RefreshAccessToken(refreshToken string) (RefreshToken, err
 			"Content-Type": {"application/x-www-form-urlencoded"},
 		},
 		Body: body,
+		AddClientAuthorization: true,
 	})
```

Comment:
Is there a reason why you did not just add `SetBasicAuth` on a request here? It would be nice to have a single way of setting the auth header.

Also, would be nice to have a test for this in `refresh_token_test.go`.